ni_measurement_plugin_sdk_service/_internal/parameter/encoder.py

```diff
@@ -0,0 +1,222 @@
+# enums and default values
+from enum import Enum
+from typing import Any, Dict, Sequence
+from uuid import uuid4
+
+from google.protobuf import descriptor_pb2, descriptor_pool, message_factory
+from google.protobuf.descriptor_pb2 import FieldDescriptorProto
+
+from ni_measurement_plugin_sdk_service._internal.parameter.decoder_strategy import get_type_default
+from ni_measurement_plugin_sdk_service._internal.parameter.metadata import (
+    ParameterMetadata,
+)
+from tests.utilities.stubs.loopback.types_pb2 import ProtobufColor
+
+
+def serialize_parameters(
+    parameter_metadata_dict: Dict[int, ParameterMetadata],
+    parameter_values: Sequence[Any],
+) -> bytes:
+    """Serialize the parameter values in same order based on the metadata_dict.
+
+    Args:
+        parameter_metadata_dict (Dict[int, ParameterMetadata]): Parameter metadata by ID.
+
+        parameter_value (Sequence[Any]): Parameter values to serialize.
+
+    Returns:
+        bytes: Serialized byte string containing parameter values.
+    """
+    pool = descriptor_pool.Default()
+    file_descriptor_proto = descriptor_pb2.FileDescriptorProto()
+    original_guid = str(uuid4())
+    unique_descriptor_name = str("msg" + "".join(filter(str.isalnum, original_guid))[:16])
```

Comment:
I think it's something we came up with, Joel commented I should changed from 'new_guid' to 'unique_descriptor_name'. Do you want me to change it back? Or remove the 'unique' part of the name?

Reply:
I think we may need to pass more information into these functions so that they can reuse the same descriptor proto rather than creating a new one each time.